### PR TITLE
feat(zimbra): preparation feature add alias

### PR DESCRIPTION
**Pull request by ghyenne**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/MANAGER-14926` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-14927 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description
It's a preparation for feature add alias

### CODE REVIEW

**Review by qpavy — APPROVED** [review]

**Review by MaximeBajeux — APPROVED** [review]

File: packages/manager/apps/zimbra/src/pages/dashboard/EmailAccounts/ModalAddAlias.tsx [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+    {
+      queryKey: getZimbraPlatformAccountDetailQueryKey(
+        platformId,
+        editEmailAccountId,
+      ),
+      queryFn: () =>
+        getZimbraPlatformAccountDetail(platformId, editEmailAccountId),
+      enabled: !!platformId && !!editEmailAccountId,
+    },
+  );
```
Comment (anooparveti) [inline packages/manager/apps/zimbra/src/pages/dashboard/EmailAccounts/ModalAddAlias.tsx:84]: create a hook for this.

File: packages/manager/apps/zimbra/src/pages/dashboard/EmailAccounts/ModalAddAlias.tsx [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+
+  interface FormTypeInterface {
+    [key: string]: FieldType;
+  }
+
+  interface FormInputRegexInterface {
+    [key: string]: RegExp;
+  }
+
+  const formInputRegex: FormInputRegexInterface = {
```
Comment (anooparveti) [inline packages/manager/apps/zimbra/src/pages/dashboard/EmailAccounts/ModalAddAlias.tsx:54]: **nitpick:** This should be moved to constant file.

File: packages/manager/apps/zimbra/src/pages/dashboard/EmailAccounts/ModalAddAlias.tsx [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+            level={ODS_THEME_TYPOGRAPHY_LEVEL.body}
+            hue={ODS_THEME_COLOR_HUE._500}
+          >
+            {t('zimbra_account_alias_add_error_message', {
+              error: response.data.message,
+            })}
+          </OsdsText>,
+          true,
+        );
+      });
```
Comment (anooparveti) [inline packages/manager/apps/zimbra/src/pages/dashboard/EmailAccounts/ModalAddAlias.tsx:134]: **nitpick:** Handle this tanstack query mutation.

File: packages/manager/apps/zimbra/src/pages/dashboard/EmailAccounts/ModalAddAlias.tsx (hunk 1 above)
Comment (ghyenne) [reply packages/manager/apps/zimbra/src/pages/dashboard/EmailAccounts/ModalAddAlias.tsx]: It's applied

File: packages/manager/apps/zimbra/src/pages/dashboard/EmailAccounts/ModalAddAlias.tsx (hunk 3 above)
Comment (ghyenne) [reply packages/manager/apps/zimbra/src/pages/dashboard/EmailAccounts/ModalAddAlias.tsx]: It's applied

File: packages/manager/apps/zimbra/src/pages/dashboard/EmailAccounts/ModalAddAlias.tsx (hunk 2 above)
Comment (ghyenne) [reply packages/manager/apps/zimbra/src/pages/dashboard/EmailAccounts/ModalAddAlias.tsx]: It's applied

**Review by anooparveti — APPROVED** [review]

**Pull request merged**